### PR TITLE
Pre-commit CI for code quality checks and auto-fix

### PR DESCRIPTION
**Pull request by Harshith-umesh**
### Description

This pre-commit ci is inspired from ryan's pre-commit ci work, this ci adds linting hooks which helps to maintain code quality. Various linter hooks have been made use of. This PR fixes the code quality of the current code base in snappy-data-server repository. Another new additon is the use of pre-commit CI service over github workflows. 
In ryan's words "The advantage of the pre-commit CI service over our current GitHub workflow is that the pre-commit CI will run pre-commit on new PRs and commits and if any tools make changes during the run, it will create a PR with those changes. This helps keep our code consistent with our pre-commit configuration, while reducing the amount of manual work we need to do to fix our code against our hooks. Additionally, the pre-commit CI is faster and automatically updates hook versions within our configuration. More details can be found on their homepage: https://pre-commit.ci. "

Examples of the ci in action:

Auto-update PR: https://github.com/Harshith-umesh/snappy-data-server/pull/2
Auto fix commit by pre-commit-ci for the above PR: https://github.com/Harshith-umesh/snappy-data-server/commit/015ee7c646831b4991dd8c6e5ba9ebf1589c8259
pre-commit-ci results : https://results.pre-commit.ci/run/github/357074993/1629890996.V53ttdBtSDuJ3bzsJdbKdQ

### CODE REVIEW

**Review by learnitall — APPROVED** [review]
Looks good! :+1:

File: app/app/main.py [hunk 1]
```diff
@@ -1,70 +1,87 @@
-import os, shutil, time
+import json
+import logging
+import os
+import shutil
+import time
+from datetime import date
 from pathlib import Path
 from typing import Any, Dict, Iterable, Optional
```
Comment (mfleader) [inline app/app/main.py:8]: Shouldn't the linter have pointed these out as not being used?

**Review by mfleader — APPROVED** [review]

File: app/app/main.py (hunk 1 above)
Comment (Harshith-umesh) [reply app/app/main.py:8]: they're used in the pruner code

**Review by amitsagtani97 — APPROVED** [review]
LGTM!

**Pull request merged**